### PR TITLE
add an ability to manage machine volumes in UD

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Add an ability to use additional volumes in machines.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/7478
https://github.com/eclipse/che/issues/7865

#### Release Notes

 - add an ability to manage machine volumes in UD

#### Docs PR

image

selection_046
selection_047

### CODE REVIEW

**Comment by olexii4** [comment]
ci-test

**Review by ashumilova — APPROVED** [review]

**Review by akurinnoy — APPROVED** [review]

**Comment by ghost** [comment]
I'll cover the docs in the new che6 docs that I'll soon push to a branch

**Review by slemeur — APPROVED** [review]

**Comment by olexii4** [comment]
ci-test

**Review by benoitf — APPROVED** [review]

File: dashboard/src/app/workspaces/workspace-details/workspace-machine-volumes/edit-volume-dialog/edit-volume-dialog.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        <div ng-message="maxlength">The name has to be less than 20 characters long.</div>
+        <div ng-message="customValidator">This name is already in use.</div>
+      </che-input-box>
+      <che-label-container che-label-name="Path" class="edit-volume-path">
+          <che-textarea che-form="VolumeForm"
+                        id="volume-path-input"
+                        che-name="path"
+                        che-place-holder="Path in the machine *"
+                        ng-model="editMachineVolumeDialogController.path"
+                        ng-pattern="/^([\/]{1}([a-z0-9]+)?)*$/"
```
Comment (ashumilova) [inline dashboard/src/app/workspaces/workspace-details/workspace-machine-volumes/edit-volume-dialog/edit-volume-dialog.html]: fix the regexp - need to allow Upper case, - , dot (for hidden files)

**Review by ashumilova — CHANGES_REQUESTED** [review]

File: dashboard/src/app/workspaces/workspace-details/workspace-machine-volumes/edit-volume-dialog/edit-volume-dialog.html (hunk 1 above)
Comment (olexii4) [reply dashboard/src/app/workspaces/workspace-details/workspace-machine-volumes/edit-volume-dialog/edit-volume-dialog.html]: done

**Review by ashumilova — APPROVED** [review]

**Pull request merged**